lesson_8/lesson-8-rusu-daniel/Tema_8.py

```diff
@@ -0,0 +1,74 @@
+class Vehicle(object):
+    
+    def __init__(self,model,transmission,weight,cubic_capacity,number_of_seats,consumtion):
+        self.transmission = transmission
+        self.weight = weight
+        self.cubic_capacity = cubic_capacity
+        self.number_of_seats = number_of_seats
+        self.consumtion = consumtion
+        self.model = model
+        
+    def is_vehicle(self, print_result=False):
+        if not print_result:
+            return self.number_of_seats > 0 and self.number_of_seats < 9 and self.weight > 1000 and self.cubic_capacity > 1.0 and self.transmission == 'Automatic' or self.transmission == 'Manual' or self.transmission =='Semi-Automatic'
+        # cum pot sa fac aici sa returneze number_of_seats doar pentru numere intregi mai mari ca 0
+    @staticmethod
+    def message():
+        return f'This is my favorite vehicle'
+      
+class Sport_Car(Vehicle):
+    
+    def __init__(self,model,transmission,weight,cubic_capacity,number_of_seats,consumtion,acceleration,horse_power,top_speed,torque,price):
+        super(Sport_Car, self).__init__(model,transmission,weight,cubic_capacity,number_of_seats,consumtion)
+        self.acceleration = acceleration
+        self.horse_power = horse_power
+        self.price = price
+        self.top_speed= top_speed
+        self.torque = torque
+        self.price = price
+        
+        
+    def is_sport_car(self,sport_car=False):
+        if not sport_car:
+            return self.cubic_capacity > 3 and self.acceleration > 4 and self.horse_power > 250 and self.top_speed > 210 and torque > 300 and self.consumtion > 17
```

Comment:
daca aplici abordarea cu metodele pentru diferitele checkuri in clasa Vehicle aici o sa poti suprascrie ce fac metodele respective pentru SportCar si sa folosesti tot `all()`